xmppserver/src/main/java/org/jivesoftware/util/cache/ExternalizableUtil.java

```diff
@@ -35,14 +41,37 @@
  */
 public class ExternalizableUtil {
 
-    private static ExternalizableUtil instance = new ExternalizableUtil();
+    private static ExternalizableUtil instance;
 
     private ExternalizableUtilStrategy strategy = new DefaultExternalizableUtil();
 
     static {
         instance = new ExternalizableUtil();
     }
 
+    /**
+     * Pool of SAX Readers. SAXReader is not thread safe so we need to have a pool of readers.
+     */
+    private final static int POOL_SIZE = 10;
+    private final BlockingQueue<SAXReader> xmlReaders = new LinkedBlockingQueue<>(POOL_SIZE);
```

Comment:
This using a BlockingQueue would make interaction a blocking operation. I'm unsure if having a small upper-bound sized pool is the best strategy to have (as it could introduce some non-trivial blocking, becoming a bottle-neck. Should we make the upper limit equal to the amount of processes that can reasonably be expected to be concurrent (amount of CPU cores? Worker threads?). In any case, making this value configurable seems sensible.